### PR TITLE
Factory Reset

### PR DESCRIPTION
**Pull request by sagarjminim**
SW-2878

### CODE REVIEW

File: patches/minim/0005-r14-Reset-indication-in-power-led.patch [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+index 2403122ad2..43ffae0a18 100755
+--- a/package/base-files/files/etc/rc.button/reset
++++ b/package/base-files/files/etc/rc.button/reset
+@@ -7,8 +7,9 @@ OVERLAY="$( grep ' /overlay ' /proc/mounts )"
+ case "$ACTION" in
+ pressed)
+ 	[ -z "$OVERLAY" ] && return 0
+-
+-	return 5
++	sleep 10
```
Comment (richp-minim) [inline patches/minim/0005-r14-Reset-indication-in-power-led.patch]: I'm not sure about this first sleep. If a user presses the button, but releases it before the 10 seconds is up, then we'd get stuck in here and start flashing the LEDs after the sleep ends, even though the user didn't hold long enough for factory reset condition.

Comment (Tom-Keddie) [reply patches/minim/0005-r14-Reset-indication-in-power-led.patch:21]: That sleep is unreachable.

Should just return 10 and the architecture will take care of the timing.

Comment (ghost) [reply patches/minim/0005-r14-Reset-indication-in-power-led.patch]: @sagarjminim take a look at https://openwrt.org/docs/guide-user/hardware/hardware.button, you're doing the work that the infrastructure does for you.  Take a look at the 7600 and do the same here.

image

File: patches/minim/0005-r14-Reset-indication-in-power-led.patch [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ 	. /etc/diag.sh
+@@ -20,7 +21,7 @@ released)
+ 		echo "REBOOT" > /dev/console
+ 		sync
+ 		reboot
+-	elif [ "$SEEN" -ge 5 -a -n "$OVERLAY" ]
++	elif [ "$SEEN" -ge 10 -a -n "$OVERLAY" ]
+ 	then
+ 		echo "FACTORY RESET" > /dev/console
+ 		jffs2reset -y && reboot &
```
Comment (ghost) [inline patches/minim/0005-r14-Reset-indication-in-power-led.patch]: We wanted this line changed to call the script in /sbin in the same way that the agent does the factory reset.

**Review by ghost — CHANGES_REQUESTED** [review]

File: patches/minim/0005-r14-Reset-indication-in-power-led.patch (hunk 2 above)
Comment (sagarjminim) [reply patches/minim/0005-r14-Reset-indication-in-power-led.patch]: Working on it.

File: patches/minim/0005-r14-Reset-indication-in-power-led.patch (hunk 1 above)
Comment (sagarjminim) [reply patches/minim/0005-r14-Reset-indication-in-power-led.patch]: Working on it.

File: patches/minim/0005-r14-Reset-indication-in-power-led.patch (hunk 2 above)
Comment (sagarjminim) [reply patches/minim/0005-r14-Reset-indication-in-power-led.patch]: @Tom-Keddie 
In the recent commit, I am not using the released case as I took the reference from 7600.

**Review by richp-minim — APPROVED** [review]

**Review by sagarjminim — COMMENTED** [review]
@Tom-Keddie Please review the changes.

**Comment by sagarjminim** [comment]
@Tom-Keddie @richp-minim @bmastbergen 
When I am creating the new branch from mh7020-v2.6.0-minim the changes in the reset file are already there. I am confused without merging the branch how are they reflected?

**Pull request closed without merging**